public/js/SceneManager.js

```diff
@@ -52,9 +53,96 @@ class SceneManager {
     // current level layout and its tilesand tile
     this.currentLevel = LEVEL_ONE;
 
+    //music menu
+    this.musicMenu = new MusicMenu(this.gameEngine, 1050, 50, [
+      this.gameEngine.assetManager.getAsset('../img/music_menu.png'),
+      this.gameEngine.assetManager.getAsset('../img/music_menu_xmusic.png'),
+      this.gameEngine.assetManager.getAsset('../img/music_menu_xfx.png')
+    ]);
+
+  }
+
+  update() {
+    this.musicMenu.update();
   }
 
+  draw() {
+    this.musicMenu.draw();
+  }
+}
+
+/**
+* Class that holds a music menu to control the play and pause music backgrounds
+* and the sound fx. Also can pause and play music using the sound engine. 
+*/
+class MusicMenu {
+  constructor(game, xPosition, yPosition, images) {
+    this.gameEngine = game;
+    this.x = xPosition;
+    this.y = yPosition;
+    this.menu = images[0];
+    this.xMusic = images[1];
+    this.xFx = images[2];
 
+    this.ctx = this.gameEngine.ctx //not needed
+    this.xMusicChecked = false;
+    this.xFxChecked = false;
+
+  }
 
+  update() {
+    let clickXandY = this.gameEngine.click;
+    let changeInState = false;
+    if (clickXandY != null) {
+      // console.log(clickXandY);
+      //music x
+      if (
+        clickXandY.x >= this.x + 10 &&
+        clickXandY.x <= this.x + 70 &&
+        clickXandY.y >= this.y + 10 &&
+        clickXandY.y <= this.y + 20
+      ) {
+        changeInState = true;
+        this.xMusicChecked = this.xMusicChecked ? false : true;
+        // console.log("music checked", this.xMusicChecked);
+      }
+      //fx x
+      if (
+        clickXandY.x >= this.x + 10 &&
+        clickXandY.x <= this.x + 70 &&
+        clickXandY.y >= this.y + 30 &&
+        clickXandY.y <= this.y + 50
+      ) {
+        changeInState = true;
+        this.xFxChecked = this.xFxChecked ? false : true;
```

Comment:
could you just make these `this.xFxChecked = !this.xFxChecked`?, although I do like the ternary operator:)

Reply:
That makes more sense. I'll change it up

Reply:
Well I'm not sure now we can figure it out tonight lol